source/src/ModuleApplication.cc

```diff
@@ -406,30 +485,114 @@ namespace dqm4hep {
     
     //-------------------------------------------------------------------------------------------------
     
+    void ModuleApplication::configureEventReader(core::TiXmlElement *element) {
+      if(EVENT_READER != appRunningMode()) {
+        return;
+      }
+      std::string eventReaderName, eventFileName;
+      int skipNEvents = 0;
+      core::TiXmlHandle handle(element);
+      THROW_RESULT_IF(core::STATUS_CODE_SUCCESS, !=, core::XmlHelper::readParameter(handle, "EventReader", eventReaderName));
+      m_eventReader = core::PluginManager::instance()->create<core::EventReader>(eventReaderName);
+      if(nullptr == m_eventReader) {
+        dqm_error( "Could not found event reader {0} in list of plugins", eventReaderName );
+        throw core::StatusCodeException(core::STATUS_CODE_NOT_FOUND);
+      }
+      dqm_info( "Loaded file reader plugin '{0}'", eventReaderName );
+      THROW_RESULT_IF(core::STATUS_CODE_SUCCESS, !=, core::XmlHelper::readParameter(handle, "EventFileName", eventFileName));
+      dqm_info( "Opening file: {0}", eventFileName );
+      THROW_RESULT_IF_AND_IF(core::STATUS_CODE_SUCCESS, core::STATUS_CODE_NOT_FOUND, !=, core::XmlHelper::readParameter(handle, "SkipNEvents", skipNEvents));
+      THROW_RESULT_IF(core::STATUS_CODE_SUCCESS, !=, m_eventReader->open(eventFileName));
+      if(0 != skipNEvents) {
+        dqm_info( "Will skip {0} first events ...", skipNEvents );
+        m_eventReader->skipNEvents(skipNEvents); // running offline ...
+      }
+      enableStats(false);
```

Comment:
We could still want to monitor ram/cpu consumption from this module no?

Reply:
My idea here was that you usually want to this event reader as an "offline mode" somehow. That's why I have disabled it. But you make a good point. I actually don't know if it is relevant. A user that want to start an event would have also to start an online-manager-server to look at the statistics. I don't know if it still makes sense...